scholarx/archive/functions.js

```diff
@@ -49,12 +53,53 @@ async function getData() {
 
 async function loadData() {
     const payload = await getData();
-    const mentorProfiles = payload.mentor2021.concat(payload.mentor2020).concat(payload.mentor2019);
+    const m2019 = payload.mentor2019; const m2020 = payload.mentor2020; const m2021 = payload.mentor2021;
+    const n2019 = payload.mentee2019; const n2020 = payload.mentee2020; const n2021 = payload.mentee2021;
+
+    //save mentor mentee details in global array variables to faster data access.
+    [mentor2019,mentor2020,mentor2021,mentor20192020,mentor20202021,mentor20192021] = [m2019,m2020,m2021,m2020.concat(m2021),m2020.concat(m2021),m2019.concat(m2021)];
+    [mentee2019,mentee2020,mentee2021,mentee20192020,mentee20202021,mentee20192021] = [n2019,n2020,n2021,n2020.concat(n2019),n2020.concat(n2021),n2019.concat(n2021)];
+ 
+    const mentorProfiles = m2019.concat(m2020).concat(m2021);
     let mentor_Profiles = Mustache.render($("#templateMentors").html(), { "mentorProfiles": mentorProfiles });
     $("#mentorProfiles").html(mentor_Profiles);
 
-    const menteeProfiles = payload.mentee2021.concat(payload.mentee2020).concat(payload.mentee2019);
+    const menteeProfiles = n2019.concat(n2020).concat(n2021);
     let mentee_Profiles = Mustache.render($("#templateMentees").html(), { "menteeProfiles": menteeProfiles });
     $("#menteeProfiles").html(mentee_Profiles);
 }
 loadData();
+
+function renderData(mentorYear,menteeYear) {
+    let mentor_Profiles = Mustache.render($("#templateMentors").html(), { "mentorProfiles": mentorYear });
+    let mentee_Profiles = Mustache.render($("#templateMentees").html(), { "menteeProfiles": menteeYear });
+    $("#mentorProfiles").html(mentor_Profiles);
+    $("#menteeProfiles").html(mentee_Profiles);
+}
+
+function renderAll() {
+    const mentorProfiles = mentor2019.concat(mentor2020).concat(mentor2021);
+    const menteeProfiles = mentee2019.concat(mentee2020).concat(mentee2021);
+    let mentor_Profiles = Mustache.render($("#templateMentors").html(), { "mentorProfiles": mentorProfiles });
+    let mentee_Profiles = Mustache.render($("#templateMentees").html(), { "menteeProfiles": menteeProfiles });
+    $("#mentorProfiles").html(mentor_Profiles);
+    $("#menteeProfiles").html(mentee_Profiles);
+}
+
```

Comment:
This too
```suggestion
```

Reply:
@YoshithaRathnayake Actually yoshitha , I put a empty line after every new function. So do we need to remove it?
![image](https://user-images.githubusercontent.com/88109775/205820437-96d84a6a-661e-40e7-9d57-0b9d5b77a1aa.png)
Same for line **79**

Reply:
We don't need that right? @anjula-sack